packages/App/src/balance/statistic/WeeklyChart.tsx

```diff
@@ -21,13 +21,36 @@ interface Props {
   month: number;
 }
 
+
 const WeeklyChart: FC<Props> = ({month}) => {
   const orientation = useDeviceOrientation();
   const allTransactions = useSelector(allTransactionsArray);
   const allTransactionsByMonth = useMemo(() => {
     return allTransactions.filter(it => new Date(it.date).getMonth() === month);
   }, [allTransactions, month]);
 
+  const daysInMonth = new Date(new Date().getFullYear(), month + 1, 1);
+  daysInMonth.setDate(daysInMonth.getDate() - 1);
+
+  const allDaysInMonth: Array<{
+    DayOfWeek: number;
+    day: number;
+    x: string;
+    y: number;
+  }> = [];
+
+  for (let i = 1; i <= daysInMonth.getDate(); i++) {
```

Comment:
Let's move this logic outside the component. It can be a separate function, which we will memoize next

Reply:
I've been thinking about it. But problem is that all of those logics exist around the prop 'month', that I send from parent component, so I can't move it outside component, cuz it'll not work(

Reply:
You can provide `month` as an argument to that function